### PR TITLE
Fix packages/app jest config

### PR DESCRIPTION
**Pull request by vikr01**
It's said to `nohoist` `react-native` in [`yarn`'s documentation](https://yarnpkg.com/blog/2018/02/15/nohoist/).

There's no need for `babel.config.js` if it is not for a monorepo (in the `packages/app` and `packages/server` directories) -- just use `.babelrc.js` instead if you want it to be a javascript file.

`lerna.json` doesn't need a `packages` key if `useWorkspaces` is provided (it takes `workspaces` from `package.json`). Using both could cause unexpected behavior.

Use `require.resolve` instead of assuming where `node_modules` are. With workspaces, modules are hoisted so they *may* not be where you expect it to be.

### CODE REVIEW

**Comment by sibelius** [comment]
what is the difference from babel.config.js and .babelrc.js?

File: lerna.json [hunk 1]
```diff
@@ -1,8 +1,5 @@
 {
-  "packages": [
-    "packages/*"
-  ],
-  "version": "1.0.0",
+  "version": "independent",
```
Comment (jgcmarins) [inline lerna.json]: why this?

**Comment by sibelius** [comment]
I think we should nohoist react as well

can you bring this test to this branch https://github.com/entria/ReactNavigationRelayModern/pull/32/files?

**Comment by sibelius** [comment]
I've tried to run the example, and got this error

```jsx
 PASS   test  src/__tests__/UserList.spec.js
  ✓ render UserList (79ms)

  console.log ../../node_modules/relay-devtools/index.js:434
    Installing Relay DevTools backend. Inspect Relay Environments in this app by running `relay-devtools`. Remember to remove this in production!

  console.error ../../node_modules/react-test-renderer/cjs/react-test-renderer.development.js:6884
    The above error occurred in the <ReactRelayQueryRenderer> component:
        in ReactRelayQueryRenderer (at createQueryRendererModern.js:30)
        in QueryRendererWrapper (at UserList.spec.js:42)
        in ThemeProvider (at testUtils.js:8)
    
    Consider adding an error boundary to your tree to customize error handling behavior.
    Visit https://fb.me/react-error-boundaries to learn more about error boundaries.

  console.log src/__tests__/UserList.spec.js:48
    err:  { Configuration error:
    
    Could not locate module React mapped as:
    /Users/sibelius/Dev/entria/feedback/entria-fullstack/packages/app/node_modules/react.
    
    Please check your configuration for these entries:
    {
      "moduleNameMapper": {
        "/^React$/": "/Users/sibelius/Dev/entria/feedback/entria-fullstack/packages/app/node_modules/react"
      },
      "resolver": null
    }
        at createNoMappedModuleFoundError (/Users/sibelius/Dev/entria/feedback/entria-fullstack/node_modules/jest-resolve/build/index.js:411:17)
        at Resolver._resolveStubModuleName (/Users/sibelius/Dev/entria/feedback/entria-fullstack/node_modules/jest-resolve/build/index.js:384:19)
        at Resolver.getMockModule (/Users/sibelius/Dev/entria/feedback/entria-fullstack/node_modules/jest-resolve/build/index.js:268:31)
        at Resolver._isModuleResolved (/Users/sibelius/Dev/entria/feedback/entria-fullstack/node_modules/jest-resolve/build/index.js:344:42)
        at Resolver._getAbsolutePath (/Users/sibelius/Dev/entria/feedback/entria-fullstack/node_modules/jest-resolve/build/index.js:322:17)
        at Resolver.getModuleID (/Users/sibelius/Dev/entria/feedback/entria-fullstack/node_modules/jest-resolve/build/index.js:301:31)
        at Runtime._shouldMock (/Users/sibelius/Dev/entria/feedback/entria-fullstack/node_modules/jest-runtime/build/index.js:766:37)
        at Runtime.requireModuleOrMock (/Users/sibelius/Dev/entria/feedback/entria-fullstack/node_modules/jest-runtime/build/index.js:460:14)
        at Object.<anonymous> (/Users/sibelius/Dev/entria/feedback/entria-fullstack/packages/app/node_modules/react-native/Libraries/Text/Text.js:240:305)
        at Runtime._execModule (/Users/sibelius/Dev/entria/feedback/entria-fullstack/node_modules/jest-runtime/build/index.js:694:13) name: '' }
```

**Comment by sibelius** [comment]
updated branch here https://github.com/entria/entria-fullstack/tree/vikr01-feat/jestMulti

not sure how to update a fork pull request

File: lerna.json (hunk 1 above)
Comment (vikr01) [reply lerna.json]: [It lets you bump versions of your packages independent of each other.](https://github.com/lerna/lerna#independent-mode)

**Comment by vikr01** [comment]
@sibelius `babel.config.js` should be used when you want `babel` to run on subdirectories that have `package.json` files.

`.babelrc` (`jsonc`) and `.babelrc.js` should be used when you only want it to work on the current package and not its subdirectories.

**Comment by sibelius** [comment]
tks, awesome PR

**Pull request merged**